src/components/datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,449 @@
+// Based on: https://github.com/City-of-Helsinki/palvelutarjotin-admin/blob/release-0.1.0/src/common/components/datepicker/Datepicker.tsx
+import {
+  OnDatesChangeProps,
+  START_DATE,
+  useDatepicker,
+} from '@datepicker-react/hooks';
+import formatDate from 'date-fns/format';
+import isValidDate from 'date-fns/isValid';
+import parseDate from 'date-fns/parse';
+import { IconAngleLeft, IconAngleRight, IconCalendarPlus } from 'hds-react';
+import uniqueId from 'lodash/uniqueId';
+import React, { useState } from 'react';
+import fiLocale from 'date-fns/locale/fi';
+import InputWrapper from './InputWrapper';
+import './Datepicker.scss';
+import './InputWrapper.scss';
+import DatepickerContext from './datepickerContext';
+import Month from './Month';
+import MonthNavButton from './MonthNavButton';
+import TimesList from './TimesList';
+
+function generateUniqueId(prefix = 'datepicker-id'): string {
+  return `${prefix}-${uniqueId()}`;
+}
+
+const dateRegex = /^\d{2}\.\d{2}\.\d{4}$/;
+const datetimeRegex = /^\d{2}\.\d{2}\.\d{4} \d{2}:\d{2}$/;
+const dateFormat = 'dd.MM.yyyy';
+const datetimeFormat = `${dateFormat} HH:mm`;
```

Comment:
It would be nice if these two were in some centralized location.
Maybe under common/utils/date-time?

Reply:
- [ ] Move these formats under common/utils/date-time